p2p/peernode.go

```diff
@@ -11,9 +14,23 @@ type PeerNode struct {
 	publicKey crypto.PubKey
 	stream    inet.Stream
 	client    pb.MainRPCClient
+	writer    *bufio.Writer
 }
 
 // GetClient returns the rpc client
 func (node *PeerNode) GetClient() pb.MainRPCClient {
 	return node.client
 }
+
+// NewPeerNode creates a P2pPeerNode
+func NewPeerNode(stream inet.Stream) PeerNode {
+	return PeerNode{
+		stream: stream,
+		writer: bufio.NewWriter(stream),
+	}
+}
+
+// SendMessage sends a protobuf message to this peer
+func (node *PeerNode) SendMessage(message proto.Message) {
+	writeMessage(message, node.writer)
```

Comment:
move the write/read/process logic into this struct so that we can ban nodes that are misbehaving